### PR TITLE
Add waveshare ESP32-S2-Pico

### PR DESCRIPTION
**Pull request by bill88t**
Disclaimer: I am not affiliated with Waveshare in any way.

It boots Circuitpython and Ljinux all nice and fine.
The activity led also works (breathing while waiting and blinking on file activity).

### CODE REVIEW

File: ports/espressif/boards/waveshare_esp32s2_pico/board.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#define LED_PIN               9
+#define LED_STATE_ON          0
+
+//--------------------------------------------------------------------+
+// USB UF2
+//--------------------------------------------------------------------+
+
+#define USB_VID           0x2E8A
+#define USB_PID           0x101F
```
Comment (hathach) [inline ports/espressif/boards/waveshare_esp32s2_pico/board.h:47]: is this vid/pid allocated for this board to use with tinyuf2. If yes, would you mind give a reference link to vid/pid

**Review by hathach — COMMENTED** [review]
thank you for your PR, please check out the review question for vid/pid

File: ports/espressif/boards/waveshare_esp32s2_pico/board.h (hunk 1 above)
Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/board.h:47]: I used the vid and pid from:
https://github.com/adafruit/circuitpython/blob/main/ports/raspberrypi/boards/waveshare_rp2040_zero/mpconfigboard.mk

As I could not find "Waveshare Electronics" here:
https://www.usb.org/sites/default/files/vendor_ids063022_0.pdf
Or in the manufacturer's wiki:
https://www.waveshare.com/wiki/ESP32-S2-Pico

The demo codes they [provide](https://www.waveshare.com/w/upload/6/69/ESP32-S2-Pico_Code.zip) use other board's firmwares and uf2's, so I can't use those as a ref...

I do not have any experience with this and I do not know what to do other than to provide waveshare_RP2040_zero's id's.

Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/board.h:47]: I have created a support ticket to waveshare.

Comment (hathach) [reply ports/espressif/boards/waveshare_esp32s2_pico/board.h:47]: thanks, let wait for the confirmation, in general, each product should have its own vid/pid.

Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/board.h]: Today, I recieved the reply.
USB_VID: 239a
USB_PID: 00a5
I have commited them to the pr.

**Comment by bill88t** [comment]
The issue has been resolved.

**Comment by bill88t** [comment]
Turns out it wasn't resolved. The vid/pid Waveshare provided were stolen from Adafruit.
Please refer to https://github.com/adafruit/circuitpython/pull/6672#discussion_r935896718

As for the new pid/vid, it was provided by [Espressif](https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt)

**Review by hathach — APPROVED** [review]
thank you very much for the PR.

**Pull request merged**